api/api.go

```diff
@@ -221,11 +221,10 @@ func NewServerRoutes(s Server) *mux.Router {
 	m.Handle(lib.AEHealth.String(), s.NoLogMiddleware(HealthCheckHandler))
 	m.Handle(lib.AEIPFS.String(), s.Middleware(s.HandleIPFSPath))
 
-	proh := NewProfileHandlers(s.Instance, cfg.API.ReadOnly)
-	m.Handle(lib.AEMe.String(), s.Middleware(proh.ProfileHandler))
-	m.Handle(lib.AEProfile.String(), s.Middleware(proh.ProfileHandler))
-	m.Handle(lib.AEProfilePhoto.String(), s.Middleware(proh.ProfilePhotoHandler))
-	m.Handle(lib.AEProfilePoster.String(), s.Middleware(proh.PosterHandler))
+	m.Handle(lib.AEGetProfile.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "profile.getprofile"))).Methods(http.MethodPost)
+	m.Handle(lib.AESetProfile.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "profile.setprofile"))).Methods(http.MethodPost)
+	m.Handle(lib.AESetProfilePhoto.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "profile.setprofilePhoto"))).Methods(http.MethodPost)
+	m.Handle(lib.AESetPosterPhoto.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "profile.setposterphoto"))).Methods(http.MethodPost)
```

Comment:
Removed the `ProfilePhoto` and `PosterPhoto` getter handlers in favor of their respective setters, as per our face to face convo today

Reply:
After the last convo, not really sure. Prior to this my mental image was that we only support getters not setters as they meddle with the config.

Reply:
So I know we aren't doing config setters over rpc, but this is specifically for use w/ a frontend that needs an api for setting any profile photos.

@b5 specific questions: in our convo on friday we talked about how if a user wants to get a config photo, they should take the path from the profile (rather than fetching the actual file from the backend). The spec still has `/profile/photo` and `/profile/poster` endpoints. Should those be the "setter" endpoints or would something link `/profile/photo/set` and `/profile/poster/set` be preferred?

Reply:
`/profile/photo` & `/profile/poster` should be POST-only, RPC-style setter endpoints.

For this pass of the API we're sticking with the "RPC can't change config" until the dispatch pattern is worked into the entire codebase, so I'd be fine with either:

* wire up these endpoints but always have them return an error
* use `denyRPC` and just turn them off entirely